### PR TITLE
Added new mapflag: MF_INSTAKILL

### PR DESCRIPTION
**Pull request by jezznar**
* **Addressed Issue(s)**: N/A

* **Server Mode**: Pre-re/Re

Had problems with tons of conflicts that broke the previous PR for MF_INSTAKILL.
Fixed.

* **Description of Pull Request**: 
Please check: https://rathena.org/board/topic/111199-release-instakill-mapflag/

**Purpose**
Create a mapflag for instakill. As it would suggest, all damages done within the map will be amplified to max damage, thus instakilling the target.
It works for the following use cases:
1. Player attacking a Monster
2. Player attacking another Player
3. Monster attacking a Player
4. Monster attacking another monster
Where "Monster" can be a normal mob or a Homunculus

**Usage**
In-game: @mapflag instakill [0|1]
Script as mapflag: prontera"tab"mapflag"tab"instakill
Script as used in setmapflag: setmapflag .EventMap$, mf_instakill;

### CODE REVIEW

**Comment by aleos89** [comment]
Can you add it to doc/mapflags.txt as well?

**Comment by jezznar** [comment]
@Jeybla I LOL'd after I read  again. My grammars got smashed there. XD

File: doc/mapflags.txt [hunk 1]
```diff
@@ -428,8 +428,7 @@ Ignores config value of 'monster_hp_bars_info'.
 
 *instakill
 
-Outputs maximum damage when attacking/receiving a/from players or monsters.
-Ignores any damage reduction.
-Attacks may either be a skill or a normal attack.
+Applies maximum damage on every hit or skill from players or monsters.
+Ignores damage reduction.
```
Comment (aleos89) [inline doc/mapflags.txt]: Could you move this to the `Battle Related` section of the documentation?

**Comment by aleos89** [comment]
The staff had a discussion about this PR. We came to the conclusion that even though this is a minimal change request, it doesn't fit into the official aspect of the emulator. As you have linked in your main post, we are happy that you released it as a diff for others to use easily. In the mean time we decided that this would be better suited on C++ rAthena plugin system. We will gladly accept this as a plugin on the C++ branch when the time comes!
Thanks and sorry again for making you go through all the review changes!

**Comment by jezznar** [comment]
noted and thank you.

**Pull request closed without merging**